### PR TITLE
Add support for multiple deployment targets with Xcode 14

### PR DESCRIPTION
**Pull request by Dmitry-Pliushchai**
### Short description 📝
Xcode 14 added multiplatform support . It means that target can support multiple platforms that quite powerfull especially if you need to support multiple platforms for your custom framework. It inspired me to add multiplatform support in tuist

### How to test the changes locally 🧐
You can use multi_platforms_app  in  /projects/tuist/fixtures/ to follow the changes

### Checklist ✅

- [x] The code architecture and patterns are consistent with the rest of the codebase
- [x] The title of the PR will be used as changelog entry, please make sure it is clear and suitable
- [x] In case the PR introduces changes that affect users, the documentation has been updated
- [x] Contributors have checked that, if needed, the PR includes the label `changelog:added`, `changelog:fixed`, or `changelog:changed`

### CODE REVIEW

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift [hunk 1]
```diff
@@ -535,7 +535,7 @@ extension ProjectDescription.Target {
                 .replacingOccurrences(of: "-", with: "_"),
             bundleId: target.name
                 .replacingOccurrences(of: "_", with: "-"),
-            deploymentTarget: deploymentTarget,
+            deploymentTargets: [deploymentTarget],
```
Comment (danyf90) [inline Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:538]: here we should use multiple deployment targets as well

File: Sources/TuistAutomation/Utilities/XcodeBuildDestination+Find.swift [hunk 2]
```diff
@@ -48,6 +48,7 @@ extension XcodeBuildDestination {
             )
             return .device(deviceAndRuntime.device.udid)
         case .macOS:
+            print("1111")
```
Comment (danyf90) [inline Sources/TuistAutomation/Utilities/XcodeBuildDestination+Find.swift:51]: this should be removed

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 3]
```diff
@@ -284,35 +285,36 @@ final class ConfigGenerator: ConfigGenerating {
...
+                    settings["SUPPORTS_MACCATALYST"] = "YES"
+                    settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
+                } else {
+                    // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
+                    settings["SUPPORTS_MACCATALYST"] = "NO"
+                }
+            case let .macOS(version):
+                settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
+            case let .watchOS(version):
+                deviceFamilyValues.append(4)
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:309]: What are these integers? maybe we should define an enum to better represent this?

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 4]
```diff
@@ -335,4 +337,22 @@ final class ConfigGenerator: ConfigGenerating {
...
+    
+    private func supportedPlatformsDerivedSettings(target: Target) -> SettingsDictionary {
+        var settings: SettingsDictionary = [:]
+        var supportedPlatforms = [String?]()
+        
+        for deploymentTarget in target.deploymentTargets {
+            guard let platform = Platform(rawValue: deploymentTarget.platform.lowercased()) else { continue }
+        
+            supportedPlatforms.append(platform.xcodeDeviceSDK)
+            supportedPlatforms.append(platform.xcodeSimulatorSDK )
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:349]:
```suggestion
            supportedPlatforms.append(platform.xcodeSimulatorSDK)
```

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 5]
```diff
@@ -109,6 +109,8 @@ public class GraphLinter: GraphLinting {
     }
 
     private func lintDependency(from: GraphTarget, to: GraphTarget) -> [LintingIssue] {
+        // Update linter to support multi platform target
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/GraphLinter.swift:112]:
```suggestion
        // TODO: Update linter to support multi platform target
```

File: Sources/TuistGenerator/Linter/SettingsLinter.swift [hunk 6]
```diff
@@ -67,11 +67,10 @@ final class SettingsLinter: SettingsLinting {
         )
 
         switch deploymentTarget {
-        case .iOS: if platform != .iOS { return [issue] }
-        case .macOS: if platform != .macOS { return [issue] }
         case .watchOS: if platform != .watchOS { return [issue] }
-        case .tvOS: if platform != .tvOS { return [issue] }
+        default: break
         }
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/SettingsLinter.swift:72]: Can you explain this?

File: Sources/TuistKit/Services/RunService.swift [hunk 7]
```diff
@@ -116,16 +116,17 @@ final class RunService {
         )
 
         let minVersion: Version?
-        if let deploymentTarget = graphTarget.target.deploymentTarget {
+        // to Update
```
Comment (danyf90) [inline Sources/TuistKit/Services/RunService.swift:119]:
```suggestion
        // TODO: Update
```

File: Sources/TuistLoader/Models+ManifestMappers/Platform+ManifestMapper.swift [hunk 8]
```diff
@@ -19,4 +19,23 @@ extension TuistGraph.Platform {
...
+            case .iOS:
+                platforms.append(.iOS)
+            case .tvOS:
+                platforms.append(.tvOS)
+            case .watchOS:
+                platforms.append(.watchOS)
+            }
+        }
+        
+        return platforms
```
Comment (danyf90) [inline Sources/TuistLoader/Models+ManifestMappers/Platform+ManifestMapper.swift:39]: Don't we have a from for the elements already?
```suggestion
        return manifest.deploymentTargets.map { .init(from: $0) }
```

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 9]
```diff
@@ -284,35 +285,36 @@ final class ConfigGenerator: ConfigGenerating {
...
             }
-        case let .macOS(version):
-            settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
-        case let .watchOS(version):
-            settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .tvOS(version):
-            settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
         }
+        
+        settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:317]: So this is what makes the device multi-platform?

**Review by danyf90 — CHANGES_REQUESTED** [review]
@kwridan I think you are the most experienced to give feedback on this part 🙏

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift (hunk 9 above)
Comment (dpliushchaiIOS) [reply Sources/TuistGenerator/Generator/ConfigGenerator.swift:317]: No, It doesn't. it defines devices/simulators that we can select as destination in Xcode. **SUPPORTED_PLATFORMS**    allows multiple platforms for target . for example SUPPORTED_PLATFORMS = "iphoneos iphonesimulator macosx";

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift (hunk 3 above)
Comment (dpliushchaiIOS) [reply Sources/TuistGenerator/Generator/ConfigGenerator.swift:309]: 1/2/3/4 - iphone/ipad/tv/watch

File: Sources/TuistGenerator/Linter/SettingsLinter.swift (hunk 6 above)
Comment (dpliushchaiIOS) [reply Sources/TuistGenerator/Linter/SettingsLinter.swift:72]: I removed check for ios/macos/tvOS because each platform from [ios, macOS, tvOS] can support any combination from [ios, macOS, tvOS]. For example: Platform can be  tvOS but it can support only macOS. **SDKROOT = appletvos SUPPORTED_PLATFORMS = macosx.**

File: Sources/ProjectDescription/Target.swift [hunk 10]
```diff
@@ -17,8 +17,8 @@ public struct Target: Codable, Equatable {
     /// The product bundle identifier.
     public let bundleId: String
 
-    /// The minimum deployment target your product will support.
-    public let deploymentTarget: DeploymentTarget?
```
Comment (kwridan) [inline Sources/ProjectDescription/Target.swift:21]: Note this is a breaking change and will require a major release

suggestion:

- Add an alternate initializer which can proxy the old API to the new one to maintain compatibility with both

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift (hunk 3 above)
Comment (kwridan) [reply Sources/TuistGenerator/Generator/ConfigGenerator.swift:309]: This information is already available in https://github.com/tuist/XcodeProj/blob/main/Sources/XcodeProj/Utils/BuildSettingsProvider.swift#L166 which Tuist uses 

I think it's worth considering an alternate approach where for each platform  we merge / append the default settings.

This also ensure we continue to respect the `defaultSettings` enum where users can opt in our out of Tuist adding any of those

File: Sources/TuistGenerator/Linter/SettingsLinter.swift (hunk 6 above)
Comment (kwridan) [reply Sources/TuistGenerator/Linter/SettingsLinter.swift:72]: Thanks for the additional info @dpliushchaiIOS - while that is possible in Xcode, would that result in a valid building target? Could the lint rules perhaps relax only when detecting valid multi platform combinations rather than none at all? I worry relaxing them here would prevent catching simple case errors which may be more common than the advanced multi platform use cases.

**Review by kwridan — COMMENTED** [review]
Thanks for starting this long standing missing feature in tuist @dpliushchaiIOS 👍 

I feel the `deploymentTarget` and `platform` APIs are a bit competing / confusing, I wonder if removing one of them if we're considering breaking changes to simplify the API would help? 

Can we trouble you to add a small fixture to `projects/tuist/fixtures/` to help demo this and allow us to test & follow along the changes in this PR? 🙂

**Comment by danyf90** [comment]
Hey @dpliushchaiIOS  👋 
No pressure at all, but did you had a look at the PR comments? 🙏

File: Sources/ProjectDescription/Target.swift (hunk 10 above)
Comment (shahzadmajeed) [reply Sources/ProjectDescription/Target.swift:21]: I agree with this approach

**Comment by dpliushchaiIOS** [comment]
> Hey @dpliushchaiIOS 👋
> No pressure at all, but did you had a look at the PR comments? 🙏

Hi. Yes, I did, just was busy last time. I will procceedwith pr next week

File: Sources/ProjectDescription/Target.swift (hunk 10 above)
Comment (dpliushchaiIOS) [reply Sources/ProjectDescription/Target.swift:21]: I am not sure if we can remove platform. For **frameworks** it really doesn't matter but platform for **apps** is needed because it defines host application for tests so we can choose which platform we are going to use for running tests.

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift (hunk 3 above)
Comment (dpliushchaiIOS) [reply Sources/TuistGenerator/Generator/ConfigGenerator.swift:309]: I Will fix that

File: Sources/TuistLoader/Models+ManifestMappers/Platform+ManifestMapper.swift (hunk 8 above)
Comment (dpliushchaiIOS) [reply Sources/TuistLoader/Models+ManifestMappers/Platform+ManifestMapper.swift]: No, We don't. There is no initializer of TuistGraph.Platform from DeploymentTarget. I added missing init.

File: Sources/ProjectDescription/Target.swift (hunk 10 above)
Comment (dpliushchaiIOS) [reply Sources/ProjectDescription/Target.swift:21]: @kwridan I added multi_platforms_app to /projects/tuist/fixtures/ to follow the changes

File: Sources/TuistGenerator/Linter/SettingsLinter.swift (hunk 6 above)
Comment (dpliushchaiIOS) [reply Sources/TuistGenerator/Linter/SettingsLinter.swift]: @kwridan  It would result in a valid building target, you can check in my example app that I added to fixtures/. I am still not sure about .watchOS, still needs some testing but it seems to be possible to have a valid building target for any platform & any deploymentTarget. I mean that it might be possible to have target where SDKROOT is .watchOS but we can launch it on iphone. I am not sure how useful it will be for app but it is really powerfull for frameworks.

**Comment by alexanderwe** [comment]
Hey all, please no pressure, just wanted to ask if there is some news about the progress here. Just highly interested in this improvement :)

**Comment by MicheleMola** [comment]
Hi guys, is there any news about this improvement? Thanks 🙏

**Comment by dpliushchaiIOS** [comment]
@MicheleMola @alexanderwe Hi. I will try to address all issues & prepare pr for review next week.

**Comment by dpliushchaiIOS** [comment]
Sorry for the delay but I am still working on it. Unfortunately,  It takes much more time that I expected.

**Comment by alexanderwe** [comment]
No worries, I can imagine this being a complex topic tbh - maybe we can try to help as a community if needed.

**Review by dzmitry-antonenka — APPROVED** [review]

**Comment by alexanderwe** [comment]
I was wondering, if the support of multi platform targets is part of the target initialiser, do we also need to expose a "platformFilter" for external targets ? e.g. there can be SPM packages with binary framework targets where eventually some of these targets do not support one platform but the others. 

e.g. 

I have a multi platform target for `.iOS` and `.watchOS`. Then I declare a dependency to `.external(name: "FrameworkA")`. `FrameworkA` is a binary target of an SPM library which references an .xcframework which only supports `.iOS`. So either tuist is "smart" enough to apply the platform filter already automatically or I need to expose the platform filter to be set by the developers, e.g. `.external(name: "FrameworkA", for: .iOS)`

@Dmitry-Pliushchai @danyf90 what do you think here ?

**Comment by danyf90** [comment]
> So either tuist is "smart" enough to apply the platform filter already automatically or I need to expose the platform filter to be set by the developers, e.g. `.external(name: "FrameworkA", for: .iOS)`

The problem is that in SPM there is no information on whether a target supports a platform or not, so I guess we don't have other choices other than having to filter on the importing target

**Comment by alexanderwe** [comment]
> The problem is that in SPM there is no information on whether a target supports a platform or not, so I guess we don't have other choices other than having to filter on the importing target

That's true and very unfortunate. I think for binary targets it __could__ be possible by examining the xcframework and take a look at the different archs that are present in the framework. But for everything else I agree we would need to have the filters being defined by the importing target. 

Would this kind of API work then `.external(name:String, platforms: [Platform] = [.iOS])` ?

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift (hunk 1 above)
Comment (Dmitry-Pliushchai) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:538]: @danyf90 Could you clarify why we need to support multiple deployments targets here? I tried to add deployment targets for each platform in package (.iOS, .macOS, .tvOS, .watchOS] but I doubt that it is correct way.
```swift
extension ProjectDescription.DeploymentTarget {
    fileprivate static func from(
        platform: ProjectDescription.Platform,
        minDeploymentTargets: [ProjectDescription.Platform: ProjectDescription.DeploymentTarget],
        package: [PackageInfo.Platform],
        packageName _: String
    ) throws -> [Self] {
        guard !package.isEmpty else { return [minDeploymentTargets[platform]!] }
        
        var deploymentTargets = [Self]()
        
        for packagePlatform in package {
            // Deployment targets below the minimum one raises warnings
            let targetVersion = try Self.max(packagePlatform.version, minDeploymentTargets[platform]?.targetVersion)

            switch ProjectDescription.Platform(rawValue: packagePlatform.platformName) {
            case .iOS:
                let hasMacCatalyst = package.contains(where: { $0.platformName == "maccatalyst" })
                deploymentTargets.append(.iOS(
                    targetVersion: targetVersion,
                    devices: hasMacCatalyst ? [.iphone, .ipad, .mac] : [.iphone, .ipad]
                )
                                         )
            case .macOS:
                deploymentTargets.append(.macOS(targetVersion: targetVersion))
            case .watchOS:
                deploymentTargets.append(.watchOS(targetVersion: targetVersion))
            case .tvOS:
                deploymentTargets.append(.tvOS(targetVersion: targetVersion))
            default:
                break
            }
        }
        
        return deploymentTargets
    }
```

File: Sources/ProjectDescription/Target.swift [hunk 11]
```diff
@@ -99,7 +99,51 @@ public struct Target: Codable, Equatable {
         self.coreDataModels = coreDataModels
         self.environment = environment
         self.launchArguments = launchArguments
-        self.deploymentTarget = deploymentTarget
+        self.deploymentTargets = deploymentTargets
         self.additionalFiles = additionalFiles
     }
+
+    public init(
```
Comment (danyf90) [inline Sources/ProjectDescription/Target.swift:106]: Let's also mark this as deprecated so people can start migrating to the new one

File: Sources/TuistGenerator/Generator/ConfigGenerator.swift [hunk 12]
```diff
@@ -284,36 +285,35 @@ final class ConfigGenerator: ConfigGenerating {
...
             }
-        case let .macOS(version):
-            settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
-        case let .watchOS(version):
-            settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .tvOS(version):
-            settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
         }
-
+
```
Comment (danyf90) [inline Sources/TuistGenerator/Generator/ConfigGenerator.swift:316]:
```suggestion

```

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 13]
```diff
@@ -108,29 +114,37 @@ public class GraphLinter: GraphLinting {
...
+                "Target \(from.target.name) has platform '\(from.target.platform)' and product '\(from.target.product)' and depends on target \(toTarget.target.name) of type \(toTarget.target.product) and platform '\(toTarget.target.platform)' which is an invalid or not supported yet combination."
+                lintingIssues.append(LintingIssue(reason: reason, severity: .error))
+                continue
+            }
+            
+            return []
         }
 
-        return []
+        return lintingIssues
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/GraphLinter.swift:147]: You can do this with a `compactMap`

File: Sources/TuistGenerator/Linter/TargetLinter.swift [hunk 14]
```diff
@@ -188,29 +188,29 @@ class TargetLinter: TargetLinting {
     }
 
     private func lintDeploymentTarget(target: Target) -> [LintingIssue] {
```
Comment (danyf90) [inline Sources/TuistGenerator/Linter/TargetLinter.swift:190]: Shouldn't we add a linter that you haven't added the same deployment target more than once?
e.g.
```
deploymentTarget: [
  .iOS("11.0"),
  .iOS("13.0"),
]
```

shouldn't be allowed

File: Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift [hunk 15]
```diff
@@ -144,10 +148,69 @@ public final class DefaultSettingsProvider: DefaultSettingsProviding {
...
+            case .iOS: return .iOS
+            case .macOS: return .macOS
+            case .watchOS: return .watchOS
+            case .tvOS: return .tvOS
+            }
+        }
+        
+        try deploymentTargets
+            .map({ mapDeploymentTarget($0) })
+            .filter({ $0 != platform })
```
Comment (danyf90) [inline Sources/TuistGenerator/Settings/DefaultSettingsProvider.swift:184]:
```suggestion
            .map { mapDeploymentTarget($0) }
            .filter { $0 != platform }
```

File: Sources/TuistKit/Services/RunService.swift [hunk 16]
```diff
@@ -116,16 +116,17 @@ final class RunService {
         )
 
         let minVersion: Version?
-        if let deploymentTarget = graphTarget.target.deploymentTarget {
+        // TODO: Update
```
Comment (danyf90) [inline Sources/TuistKit/Services/RunService.swift:119]: Update what?

File: Sources/ProjectDescription/Target.swift (hunk 10 above)
Comment (danyf90) [inline Sources/ProjectDescription/Target.swift:21]: What does it mean to have a single platform but multiple deployment targets? shouldn't we remove the platform parameter?

**Review by danyf90 — CHANGES_REQUESTED** [review]

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift (hunk 1 above)
Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:538]: Let's say you have a multi platform iOS+macOS target depending on an external package, shouldn't the target coming from the package be iOS+macOS as well?

Comment (danyf90) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift:538]: Also, right now [we are creating a target for each platform](https://github.com/tuist/tuist/blob/45b1c71bb008aaa86751de6c9ec3789fc9fd6c03/Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift#L383-L401), while we should create a single multi-platform target

File: Sources/ProjectDescription/Target.swift [hunk 17]
```diff
@@ -67,7 +67,7 @@ public struct Target: Codable, Equatable {
         product: Product,
         productName: String? = nil,
         bundleId: String,
-        deploymentTarget: DeploymentTarget? = nil,
+        deploymentTargets: [DeploymentTarget],
```
Comment (shahzadmajeed) [inline Sources/ProjectDescription/Target.swift:70]: This will be a breaking change. You probably want to default to empty array?

Comment (shahzadmajeed) [reply Sources/ProjectDescription/Target.swift:70]: Nevermind, just saw the other initializer below

File: Sources/TuistAutomation/Utilities/XcodeBuildDestination+Find.swift [hunk 18]
```diff
@@ -25,19 +25,21 @@ extension XcodeBuildDestination {
...
                 minVersion = scheme.targetDependencies()
-                    .flatMap {
+                    .compactMap {
                         graphTraverser
                             .directLocalTargetDependencies(path: $0.projectPath, name: $0.name)
                             .map(\.target)
-                            .map(\.deploymentTarget)
-                            .compactMap { $0?.version.version() }
+                            .flatMap(\.deploymentTargets)
+                            .first(where: { $0.platform.lowercased() == target.platform.rawValue } )
```
[the remark this answers is not in the document]
Comment (shahzadmajeed) [reply Sources/TuistAutomation/Utilities/XcodeBuildDestination+Find.swift:37]: Wouldn't there be a minimum version for each platform?

**Comment by danyf90** [comment]
@Dmitry-Pliushchai  can you please have a look at the failing workflows? 🙏

**Comment by Dmitry-Pliushchai** [comment]
> @Dmitry-Pliushchai can you please have a look at the failing workflows? 🙏

@danyf90 Sure, will take a look. it is weird because it worked ok locally & tests passed successfully. maybe because of some cache or so but I will figure it out.

**Comment by danyf90** [comment]
@Dmitry-Pliushchai the associated issue got assigned a bounty of 100$, so you'll receive it if you manage to bring this to the finish line! 🎉

**Comment by thedavidharris** [comment]
> @Dmitry-Pliushchai the associated issue got assigned a bounty of 100$, so you'll receive it if you manage to bring this to the finish line! 🎉

Is Tuist looking to drop Xcode 13 support sooner than the April 2023 deadline set by Apple? I believe that's the main holdup on this as is, given that CI is still compiling with Xcode 13

**Comment by danyf90** [comment]
> > @Dmitry-Pliushchai the associated issue got assigned a bounty of 100$, so you'll receive it if you manage to bring this to the finish line! 🎉
> 
> 
> 
> Is Tuist looking to drop Xcode 13 support sooner than the April 2023 deadline set by Apple? I believe that's the main holdup on this as is, given that CI is still compiling with Xcode 13

If you mean for developing tuist I think we can drop it now already

**Comment by thedavidharris** [comment]
> If you mean for developing tuist I think we can drop it now already

Cool, that might be the fix here.

**Comment by danyf90** [comment]
> > If you mean for developing tuist I think we can drop it now already
> 
> 
> 
> Cool, that might be the fix here. 

Feel free to open a separate PR for that if you have time 🙏

**Comment by Dmitry-Pliushchai** [comment]
> @Dmitry-Pliushchai the associated issue got assigned a bounty of 100$, so you'll receive it if you manage to bring this to the finish line! 🎉

Sure. Will continue this/next week.

File: Gemfile.lock [hunk 19]
```diff
@@ -3,24 +3,25 @@ GEM
   specs:
     CFPropertyList (3.0.5)
       rexml
-    actionpack (7.0.4)
-      actionview (= 7.0.4)
-      activesupport (= 7.0.4)
-      rack (~> 2.0, >= 2.2.0)
+    actionpack (6.1.7.2)
```
Comment (danyf90) [inline Gemfile.lock:6]: Why are we downgrading things?

Comment (Dmitry-Pliushchai) [reply Gemfile.lock:6]: After some investigation I still have no idea why bundle install/update downgrade some of gems. will investigate

File: Sources/ProjectDescription/Cache.swift [hunk 20]
```diff
@@ -41,7 +41,8 @@ public struct Cache: Codable, Equatable {
 
     /// Returns a `Cache` instance containing the given profiles.
     /// If no profile list is provided, tuist's default profile will be taken as the default.
-    /// If no profile is provided in `tuist cache --profile` command, the first profile from the profiles list will be taken as the default.
+    /// If no profile is provided in `tuist cache --profile` command, the first profile from the profiles list will be taken as
+    /// the default.
```
Comment (danyf90) [inline Sources/ProjectDescription/Cache.swift:45]: @Dmitry-Pliushchai can you clean up the PR from mall the unrelated changes like this or the gems? 🙏

File: Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift (hunk 1 above)
Comment (Dmitry-Pliushchai) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift]: @danyf90 What do you think about naming multiplatform target? In my opinion single platform target should has platform prefix but Multiplatform target shouldn't.

Comment (Dmitry-Pliushchai) [reply Sources/TuistDependencies/SwiftPackageManager/Utils/PackageInfoMapper.swift]: yes, I think we can remove prefixes at all for single multiplatform target & single platform target, can't we?

**Comment by Dmitry-Pliushchai** [comment]
@danyf90 Do you have any objections to change macOS deployment targets from 10.15 to 13 as we age going to drop Xcode13?

**Comment by shahzadmajeed** [comment]
> 

That’s a big jump. We should keep at least MacOS 12 for current users

**Comment by danyf90** [comment]
> > 
> 
> 
> 
> That’s a big jump. We should keep at least MacOS 12 for current users

Yep I would keep 12. Also, GitHub actions don't support 13 yet 😁

**Comment by EthanLipnik** [comment]
Can someone reopen the issue? It's still in progress and is important
